### ISSUE TITLE
Required `Self` is removed

### ISSUE THREAD

**Issue by kitwtnb** [nicklockwood/SwiftFormat#1503]
The following enum is defined.

```swift
enum Foo {
    static var value = 0

    static func f(value: Int) {
        Self.value = value
    }
}
```

In 0.51.15, it is not rewritten.

In 0.52.0, it was rewritten as follows, and the compilation no longer passes.

```swift
enum Foo {
    static var value = 0

    static func f(value: Int) {
        value = value
    }
}
```

**Comment by nicklockwood** [comment]
@kitwtnb thanks for flagging this, I'll get it fixed. In the meantime, you can work around it by disabling the `redundantStaticSelf` rule (cc @sjavora).

**Comment by nicklockwood** [comment]
@kitwtnb fixed in 0.52.1

**Comment by kitwtnb** [comment]
@nicklockwood   
Thank you very much.
The above pattern has been fixed, but the following pattern will now remove the required `Self`.

0.52.1  
```swift
class Foo {
    static let bar = 0
    
    func f() {
        let a = Self.bar
    }
}
```
↓  
```swift
class Foo {
    static let bar = 0

    func f() {
        let a = bar
    }
}
```

**Comment by nicklockwood** [comment]
@kitwtnb I'm not able to reproduce that with 0.52.1. Can you double check you're using the right version and that particular code snippet reproduces the issue?

**Comment by kitwtnb** [comment]
@nicklockwood 

The following steps were used to reproduce the results.

1. Write code in `./Foo.swift`.

```swift
class Foo {
    static let bar = 0
    
    func f() {
        let a = Self.bar
    }
}
```

2. Run it using Mint.  
`mint run nicklockwood/SwiftFormat@0.52.1 Foo.swift`

3. `Self` has been removed.

**Comment by nicklockwood** [comment]
@kitwtnb with that exact process and those exact file contents, I'm seeing no changes. I have found a bug involving `let` or `if let` expressions, so for the following code `Self` is currently removed by 0.52.1, but *not* for the example you gave, which seems odd.

```swift
class Foo {
    static let bar = 0

    func f() {
        if let a = Self.bar {
            print("oops")
        }
    }
}
```

**Comment by kitwtnb** [comment]
It is also reproduced in 0.52.1.

However, this has been corrected in 0.52.2.
Thank you very much!
